### DISCUSSION TITLE
JS not working as expected on first load (works after refresh)

### DISCUSSION THREAD

**Discussion by ghost** [FreshRSS/FreshRSS#5131] [Q&A]
When I try testing if an element is running, it is not working (partially or entirely) while rendering the page for the first time. Once I refreshed and ran it a second time, it worked perfectly well.
Since the page has been cached during the first load, the remaining elements will be loaded the next time. The main.js file also contains a setTimeout for the elements to be ready.
Is it advisable to place the script files at the end of the page (the body) when you cannot entirely depend upon DOM to load.

**Comment by Frenzie** [comment]
I'm not quite sure what you're asking. You can tell your extension to put the script at the bottom or you can use DOMContentLoaded.
